nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-parameter-contexts.js

```diff
@@ -692,16 +692,26 @@
                                         }
                                     } else {
                                         var referencingUnauthorizedComponentContainer = $('<li class="referencing-component-container"></li>').appendTo(unauthorizedComponentsContainer);
-                                        $('<span class="parameter-context-referencing-component-name link ellipsis"></span>').prop('title', unauthorizedReferencingComponentEntity.id).text(unauthorizedReferencingComponentEntity.id).on('click', function () {
-                                            // check if there are outstanding changes
-                                            handleOutstandingChanges().done(function () {
-                                                // close the shell
-                                                $('#shell-dialog').modal('hide');
-
-                                                // show the component in question
-                                                nfCanvasUtils.showComponent(unauthorizedReferencingComponentEntity.processGroup.id, unauthorizedReferencingComponentEntity.id);
-                                            });
-                                        }).appendTo(referencingUnauthorizedComponentContainer);
+                                        $('<span class="parameter-context-referencing-component-name link ellipsis"></span>')
+                                            .prop('title', unauthorizedReferencingComponentEntity.id)
+                                            .text(unauthorizedReferencingComponentEntity.id)
+                                            .on('click', function () {
+                                                // check if there are outstanding changes
+                                                handleOutstandingChanges().done(function () {
+                                                    // close the shell
+                                                    $('#shell-dialog').modal('hide');
+
+                                                    // show the component in question
+                                                    if (unauthorizedReferencingComponentEntity.referenceType === 'PROCESSOR') {
+                                                        nfCanvasUtils.showComponent(unauthorizedReferencingComponentEntity.processGroup.id, unauthorizedReferencingComponentEntity.id);
+                                                    } else if (unauthorizedReferencingComponentEntity.referenceType === 'CONTROLLER_SERVICE') {
+                                                        nfProcessGroupConfiguration.showConfiguration(unauthorizedReferencingComponentEntity.processGroup.id).done(function () {
+                                                            nfProcessGroupConfiguration.selectControllerService(unauthorizedReferencingComponentEntity.id);
```

Comment:
You will first need to navigate the user to the correct PG. This will update the url as well as the breadcrumbs to give the user context as to where they are.

```suggestion
nfProcessGroup.enterGroup(unauthorizedReferencingComponentEntity.processGroup.id);                                                            nfProcessGroupConfiguration.selectControllerService(unauthorizedReferencingComponentEntity.id);
```